<?xml version="1.0" ?>
<sdf version="1.5">
  <model name="soccer_ball">
    <static>false</static>
    <link name="soccer_ball_link">
      <pose>0 0 0 0 0 0</pose>
      <inertial>
        <mass>0.026</mass>
        <!-- inertia based on solid sphere 2/5 mr^2 -->
        <inertia>
          <ixx>1.664e-5</ixx>
          <iyy>1.664e-5</iyy>
          <izz>1.664e-5</izz>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyz>0</iyz>
        </inertia>
      </inertial>
      <collision name="collision">
        <geometry>
          <sphere>
            <radius>0.04</radius>
          </sphere>
        </geometry>
        <!-- FRICTION ADDED BY ME ADAM -->
        <surface>
          <bounce>
            <restitution_coefficient>0.1</restitution_coefficient>
            <threshold>0</threshold>
          </bounce>
          <contact>
            <ode>
              <max_vel>1</max_vel>
            </ode>
          </contact>
          <friction>
            <ode>
              <mu>1</mu>
              <mu2>0.5</mu2>
            </ode>
          </friction>
        </surface>
        <!-- from https://github.com/osrf/gazebo/blob/gazebo_1.9/sdf/worlds/test_friction.world
        <surface>
            <friction>
              <ode>
                <mu>50.000000</mu>
                <mu2>50.000000</mu2>
                <fdir1>0.000000 0.000000 0.000000</fdir1>
                <slip1>0.000000</slip1>
                <slip2>0.000000</slip2>
              </ode>
            </friction>
            <bounce>
              <restitution_coefficient>0.000000</restitution_coefficient>
              <threshold>100000.000000</threshold>
            </bounce>
            <contact>
              <ode>
                <soft_cfm>0.000000</soft_cfm>
                <soft_erp>0.200000</soft_erp>
                <kp>1000000000.000000</kp>
                <kd>1.000000</kd>
                <max_vel>100.000000</max_vel>
                <min_depth>0.001000</min_depth>
              </ode>
            </contact>-->
          <!--</surface>-->
      </collision>
      <visual name="visual">
        <geometry>
          <mesh>
            <scale>0.04 0.04 0.04</scale>
            <uri>model://robocup_3Dsim_ball/meshes/ball.dae</uri>
          </mesh>
        </geometry>
      </visual>
    </link>
  </model>
</sdf>
